src/components/FilterAnnotMondal/FilterAnnotModal.js

```diff
@@ -0,0 +1,172 @@
+import classNames from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import defaultTool from 'constants/defaultTool';
+import core from 'core';
+import actions from 'actions';
+import selectors from 'selectors';
+
+import Choice from 'components/Choice';
+import Button from 'components/Button';
+
+import { Swipeable } from 'react-swipeable';
+import { FocusTrap } from '@pdftron/webviewer-react-toolkit';
+
+import './FilterAnnotModal.scss';
+
+const FilterAnnotModal = () => {
+  const [isDisabled, isOpen] = useSelector(state => [
+    selectors.isElementDisabled(state, 'filterModal'),
+    selectors.isElementOpen(state, 'filterModal'),
+  ]);
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+
+  const [authors, setAuthors] = useState([]);
+  const [annotTypes, setAnnotTypes] = useState([]);
+
+  const [authorFilter, setAuthorFilter] = useState([]);
+  const [typesFilter, setTypesFilter] = useState([]);
+
+  const filterApply = () => {
+    dispatch(
+      actions.setCustomNoteFilter(annot => {
+        let type = true;
+        let author = true;
+        if (typesFilter.length > 0) {
+          type = typesFilter.includes(annot.Subject.toLowerCase().replace(/\s/g, ''));
+        }
+        if (authorFilter.length > 0) {
+          author = authorFilter.includes(annot.Author);
+        }
+        return type && author;
+      }),
+    );
+    closeModal();
+  };
+
+  const filterClear = () => {
+    dispatch(
+      actions.setCustomNoteFilter(annot => {
+        return true;
+      }),
+    );
+    setAuthorFilter([]);
+    setTypesFilter([]);
+    closeModal();
+  };
+
+  const closeModal = () => {
+    dispatch(actions.closeElement('filterModal'));
+    core.setToolMode(defaultTool);
+  };
+
+  useEffect(() => {
+    const annots = core.getAnnotationsList();
+    // set is a great way to remove any duplicate additions and ensure the unique items are present
+    // the only gotcha that it should not be used by state since not always it will trigger a rerender
+    let authorsToBeAdded = new Set();
+    let annotTypesToBeAdded = new Set();
+    annots.forEach(annot => {
+      if (core.getDisplayAuthor(annot) && core.getDisplayAuthor(annot) !== '') {
+        authorsToBeAdded.add(core.getDisplayAuthor(annot));
+      }
+      if (annot.Subject && annot.Subject !== '') {
+        // we want to lowercase it and remove spaces to get i18n translations
+        annotTypesToBeAdded.add(annot.Subject.toLowerCase().replace(/\s/g, ''));
+      }
+    });
+    setAuthors([...authorsToBeAdded]);
+    setAnnotTypes([...annotTypesToBeAdded]);
+  }, [isOpen]);
+
+  const modalClass = classNames({
+    Modal: true,
+    FilterAnnotModal: true,
+    open: isOpen,
+    closed: !isOpen,
+  });
+
+  return isDisabled ? null : (
+    <Swipeable onSwipedUp={closeModal} onSwipedDown={closeModal} preventDefaultTouchmoveEvent>
+      <FocusTrap locked={isOpen}>
```

Comment:
Looks fantastic! Only suggestion would be to re-focus the "Filter" button when the modal closes (there's a prop for that)

```suggestion
      <FocusTrap locked={isOpen} focusLastOnUnlock>
```

Reply:
It does look like that only fixes it if you use escape to close though, let me look into that quickly